Signal/src/ViewControllers/MessageMetadataViewController.swift

```diff
@@ -403,4 +428,39 @@ class MessageMetadataViewController: OWSViewController {
         let data = dataSource.data()
         UIPasteboard.general.setData(data, forPasteboardType:utiType)
     }
+
+    // MARK: - Actions
+
+    // This method should be called after self.databaseConnection.beginLongLivedReadTransaction().
+    private func updateDBConnectionAndMessageToLatest() {
+
+        AssertIsOnMainThread()
+
+        self.databaseConnection.read { transaction in
+            guard let newMessage = TSInteraction.fetch(uniqueId:self.message.uniqueId, transaction:transaction) as? TSMessage else {
+                Logger.error("\(self.TAG) Couldn't reload message.")
+                return
+            }
+            self.message = newMessage
+        }
+    }
+
+    internal func yapDatabaseModified(notification: NSNotification) {
+//        Logger.info("\(TAG) in \(#function)")
+        AssertIsOnMainThread()
+
+        let notifications = self.databaseConnection.beginLongLivedReadTransaction()
+
+// TODO: I can't figure out what the !@#$%&^* auto-generated Swift interface to this method is.
+//        guard self.databaseConnection.hasChange(forKey:message.uniqueId,
```

Comment:
@michaelkirk can you please try to autocomplete this statement?  I can't generate the swift interface for YapDBConnection.

Reply:
```
    open func hasChange(forKey key: String, inCollection collection: String, in notifications: [Notification]) -> Bool
```

I put the rest of that class here if you need it: https://gist.github.com/michaelkirk/224d32429358abf996e58b7eb31d03f8

Xcode **did** crash the first 3 times I tried it =(

Reply:
THANK YOU